config/vlan.py

```diff
@@ -42,26 +53,35 @@ def del_vlan(db, vid):
     log.log_info("'vlan del {}' executing...".format(vid))
 
     ctx = click.get_current_context()
+    vlan = 'Vlan{}'.format(vid)
 
-    if not clicommon.is_vlanid_in_range(vid):
-        ctx.fail("Invalid VLAN ID {} (1-4094)".format(vid))
+    if ADHOC_VALIDATION:
+        if not clicommon.is_vlanid_in_range(vid):
+            ctx.fail("Invalid VLAN ID {} (1-4094)".format(vid))
 
-    vlan = 'Vlan{}'.format(vid)
-    if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
-        ctx.fail("{} does not exist".format(vlan))
+        if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+            ctx.fail("{} does not exist".format(vlan))
 
-    intf_table = db.cfgdb.get_table('VLAN_INTERFACE')
-    for intf_key in intf_table:
-        if ((type(intf_key) is str and intf_key == 'Vlan{}'.format(vid)) or
-            (type(intf_key) is tuple and intf_key[0] == 'Vlan{}'.format(vid))):
-            ctx.fail("{} can not be removed. First remove IP addresses assigned to this VLAN".format(vlan))
+        intf_table = db.cfgdb.get_table('VLAN_INTERFACE')
+        for intf_key in intf_table:
+            if ((type(intf_key) is str and intf_key == 'Vlan{}'.format(vid)) or # MISSING CONSTRAINT IN YANG MODEL
+                (type(intf_key) is tuple and intf_key[0] == 'Vlan{}'.format(vid))):
+                ctx.fail("{} can not be removed. First remove IP addresses assigned to this VLAN".format(vlan))
 
-    keys = [ (k, v) for k, v in db.cfgdb.get_table('VLAN_MEMBER') if k == 'Vlan{}'.format(vid) ]
+        keys = [ (k, v) for k, v in db.cfgdb.get_table('VLAN_MEMBER') if k == 'Vlan{}'.format(vid) ]
     
-    if keys:
-        ctx.fail("VLAN ID {} can not be removed. First remove all members assigned to this VLAN.".format(vid))
+        if keys: # MISSING CONSTRAINT IN YANG MODEL
+            ctx.fail("VLAN ID {} can not be removed. First remove all members assigned to this VLAN.".format(vid))
         
-    db.cfgdb.set_entry('VLAN', 'Vlan{}'.format(vid), None)
+    config_db = ValidatedConfigDBConnector(db.cfgdb, ADHOC_VALIDATION)
+    try: 
+        config_db.set_entry('VLAN', 'Vlan{}'.format(vid), None)
```

Comment:
>None [](http://example.com/codeflow?start=58&length=4)

You should handle `set_entry` `None` as special case in decorator.

Reply:
Added special case to decorator